src/polyfem/solver/forms/FrictionForm.cpp

```diff
@@ -128,15 +130,15 @@ namespace polyfem::solver
 	{
 		const Eigen::MatrixXd displaced_surface = compute_displaced_surface(x);
 
-		ipc::CollisionConstraints constraint_set;
-		constraint_set.set_use_convergent_formulation(contact_form_.use_convergent_formulation());
-		constraint_set.set_are_shape_derivatives_enabled(contact_form_.enable_shape_derivatives());
-		constraint_set.build(
+		ipc::Collisions collision_set;
+		collision_set.set_use_convergent_formulation(contact_form_.use_convergent_formulation());
+		collision_set.set_are_shape_derivatives_enabled(contact_form_.enable_shape_derivatives());
+		collision_set.build(
 			collision_mesh_, displaced_surface, dhat_,
 			/*dmin=*/0, broad_phase_method_);
 
-		friction_constraint_set_.build(
-			collision_mesh_, displaced_surface, constraint_set,
+		friction_collision_set_.build(
+			collision_mesh_, displaced_surface, collision_set,
 			dhat_, contact_form_.barrier_stiffness(), mu_);
```

Comment:
You should replace `dhat_` here (and in the force/force jacobian) with `contact_form_.barrier_potential()` to make sure they are using the same `Barrier`.

Reply:
This compiles only because of the implicit conversion from `double` to `BarrierPotential` (the thing I mentioned to you on Slack).

Reply:
Done. Can we turn off this implicit conversion for polyfem? I feel like it could lead to a lot of bugs.

Reply:
I was wrong not even `-Werror=conversion` prevents the implicit conversion.

Reply:
Ok, I guess we just have to be careful then.